### PR TITLE
chore: ensure PR workflows can't write to GHCR

### PR DESCRIPTION
**Pull request by UncleGedd**
## Description
- ensure PR workflows can't write to GHCR (thanks @bburky !)
- create nightly workflow for testing GHCR writes

### CODE REVIEW

File: .github/workflows/nightly-ghcr.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+name: Test GHCR Write
+on:
+  schedule:
+    - cron: '0 7 * * *' ## Every day at 0700 UTC
+
+  workflow_dispatch: ## Give us the ability to run this manually
+
+permissions:
+  contents: read
```
Comment (decleaver) [inline .github/workflows/nightly-ghcr.yaml:9]: is read sufficient?

File: .github/workflows/test-e2e-pr.yaml [hunk 2]
```diff
@@ -38,18 +36,9 @@ jobs:
       - name: Setup K3d
         uses: ./.github/actions/k3d
 
-      - name: Login to GHCR
```
Comment (decleaver) [inline .github/workflows/test-e2e-pr.yaml:41]: do we not need this for the nightly?

File: .github/workflows/nightly-ghcr.yaml (hunk 1 above)
Comment (UncleGedd) [reply .github/workflows/nightly-ghcr.yaml:9]: 👍 we are enabling `write` below

File: .github/workflows/test-e2e-pr.yaml (hunk 2 above)
Comment (UncleGedd) [reply .github/workflows/test-e2e-pr.yaml]: good catch, added to nightly + `GITHUB_TOKEN` env var

**Review by decleaver — APPROVED** [review]
reviewed, looks good

**Pull request merged**